syft/serde/native_serde.py

```diff
@@ -334,9 +355,9 @@ def _detail_ndarray(
         arr = _detail_ndarray(arr_representation)
 
     """
-    res = numpy.frombuffer(arr_representation[0], dtype=arr_representation[2]).reshape(
-        arr_representation[1]
-    )
+    arr_shape = serde._detail(worker, arr_representation[1])
+    arr_dtype = serde._detail(worker, arr_representation[2])
+    res = numpy.frombuffer(arr_representation[0], dtype=arr_dtype).reshape(arr_shape)
 
     assert type(res) == numpy.ndarray
```

Comment:
Q: Shouldn't we remove this assert? We should get an assert if ```numpy.frombuffer``` will fail, right?